universal-application-tool-0.0.1/app/repository/ProgramRepository.java

```diff
@@ -110,4 +112,17 @@ public Program createOrUpdateDraft(Program existingProgram) {
       }
     }
   }
+
+  public CompletableFuture<Program> getForSlug(String slug) {
```

Comment:
(blocking) this should be stored in a column in the table so we can do an indexed search on this.

Reply:
yeah, but it will really be a hassle, and the follow-up issue is going to sit in the icebox forever... let me see if I can figure out how to do this.

Reply:
Okay, looks like I can make that work...  I'll watch the prober when this goes in.